consume.go

```diff
@@ -317,11 +352,150 @@ func mergeMessages(cs [2]<-chan *sarama.ConsumerMessage, out chan<- *sarama.Cons
 	}
 }
 
+type consumerPartitioners struct {
+	all          bool
+	partitioners []func(key []byte, numPartitions int) (int, error)
+}
+
+func parseConsumerPartitioners(ps []string, defaultPartitioner sarama.PartitionerConstructor) (*consumerPartitioners, error) {
+	var cp consumerPartitioners
+	for _, p := range ps {
+		switch p {
+		case "all":
+			cp.all = true
+		case "sarama", "std":
+			cp.partitioners = append(cp.partitioners, partitionerFunc(partitioners[p]))
+		default:
+			return nil, fmt.Errorf("unknown partitioner %q", p)
+		}
+	}
+	if cp.all {
+		// No point in having any explicit partitioners if
+		cp.partitioners = nil
+	} else if len(cp.partitioners) == 0 {
+		cp.partitioners = append(cp.partitioners, partitionerFunc(defaultPartitioner))
+	}
+	return &cp, nil
+}
+
+func partitionerFunc(makePartitioner sarama.PartitionerConstructor) func(key []byte, partitionSize int) (int, error) {
+	if makePartitioner == nil {
+		panic("bad partitioner (internal consistency error)")
+	}
+	// Note: all known partitioners ignore the topic argument to the constructor.
+	// Why would a partitioner ever behave differently depending on the topic name anyway?!
```

Comment:
It does seem a bit weird - guess the algorithm could be shared by some other code that wants to embed that data?  Pure speculation on my part.